MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,241 @@
+import Polyline
+import MapboxDirections
+import AVFoundation
+import MapboxMobileEvents
+
+let SecondsForCollectionAfterFeedbackEvent: TimeInterval = 20
+
+struct DefaultEventDictionary {
+    var originalRequestIdentifier: String?
+    var requestIdentifier: String?
+    var lat: CLLocationDegrees?
+    var lng: CLLocationDegrees?
+    var originalGeometry: String?
+    var originalEstimatedDistance: CLLocationDistance?
+    var originalEstimatedDuration: TimeInterval?
+    var geometry: String?
+    var estimatedDistance: CLLocationDistance?
+    var estimatedDuration: TimeInterval?
+    var created: String
+    var startTimestamp: String
+    var platform: String
+    var operatingSystem: String
+    var device: String
+    var sdkIdentifier: String
+    var sdkVersion: String
+    var eventVersion: Int
+    var profile: String
+    var simulation: Bool
+    var sessionIdentifier: String
+    var distanceCompleted: CLLocationDistance
+    var distanceRemaining: TimeInterval
+    var durationRemaining: TimeInterval
+    var rerouteCount: Int
+    var volumeLevel: Int
+    var screenBrightness: Int
+    var batteryPluggedIn: Bool
+    var batteryLevel: Float
+    var applicationState: String
+    
+    init(routeController: RouteController, session: SessionState) {
+        created = Date().ISO8601
+        startTimestamp = session.departureTimestamp!.ISO8601
+        
+        platform = ProcessInfo.systemName
+        operatingSystem = "\(ProcessInfo.systemName) \(ProcessInfo.systemVersion)"
```

Comment:
The operating system name and version never change, so there’s no need to store it; just add them in when we convert the struct into a dictionary. This also applies to any other state that won’t change between the time we collect the data and the time we send it.